clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/WebConfig.groovy

```diff
@@ -92,6 +93,7 @@ public class WebConfig extends WebMvcConfigurerAdapter {
   }
 
   @ControllerAdvice
+  @Order(Ordered.HIGHEST_PRECEDENCE)
   static class IllegalArgumentExceptionHandler {
     @ExceptionHandler(IllegalArgumentException)
     public void handle(HttpServletResponse response, IllegalArgumentException ex) {
```

Comment:
Would it make sense to decorate the returned error message like how it's done in [kork](https://github.com/spinnaker/kork/blob/3288c4b87be1f71178e3282d131dd815e08d3574/kork-web/src/main/java/com/netflix/spinnaker/kork/web/exceptions/GenericExceptionHandlers.java#L80) in the GenericExceptionHandler?

I'm also curious if the exception handler in [kayenta](https://github.com/spinnaker/kayenta/blob/1f43abda552afa77a11b293a87a354c6f74ebc9e/kayenta-web/src/main/java/com/netflix/kayenta/controllers/ControllerExceptionHandler.java#L32) actually works, because it doesn't have the `@Ordered` annotation.

Anyway seems that the only difference for calling code is to receive a 400 instead of 500, with the same body structure as before. This will potentially impact other services but I think this a right step forward.

Reply:
> Would it make sense to decorate the returned error message like how it's done in [kork](https://github.com/spinnaker/kork/blob/3288c4b87be1f71178e3282d131dd815e08d3574/kork-web/src/main/java/com/netflix/spinnaker/kork/web/exceptions/GenericExceptionHandlers.java#L80) in the GenericExceptionHandler?

Could be.  I haven't stared at the differences in responses by doing that.  My bigger picture goal with all of this is to try to get rosco to not retry when [this call](https://github.com/spinnaker/rosco/blob/03e4e26a6e19fd26a62842c12f93aeb8d4275d8c/rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImpl.java#L33) fails due to a misconfigured artifact.  If the decoration helps with that, I'm all for it...I'm planning to bring this up with sig-platform since it looks like there are lots of possible ways to handle this.